### PR TITLE
Make clear the log message from MailChimp response

### PR DESCRIPTION
**Pull request by thangnc**
- Renamed `MailChimpAbstractRecordBuffer` to `MailChimpRecordBuffer` and `MailChimpRecordBuffer` to `MailChimpClient` to make clear functionality as suggestion in #28 
- Make clear the log message for MailChimp error response about invalid records

### CODE REVIEW

File: src/main/java/org/embulk/output/mailchimp/MailChimpClient.java [hunk 1]
```diff
@@ -166,7 +195,7 @@ else if (task.getAuthMethod() == API_KEY && task.getApikey().isPresent()) {
                 mailchimpEndpoint = MessageFormat.format(mailchimpEndpoint, domain);
             }
         }
-        catch (JsonProcessingException jpe) {
+        catch (Exception e) {
             throw new ConfigException("Could not get data center");
```
Comment (sakama) [inline src/main/java/org/embulk/output/mailchimp/MailChimpClient.java]: `throw new ConfigException("Could not get data center", e);` is better?

**Review by sakama — COMMENTED** [review]
Add 1 comment.
Other parts looks good to me.

And let's use `git mv` to reduce diff since next time if possible.

File: src/main/java/org/embulk/output/mailchimp/MailChimpClient.java (hunk 1 above)
Comment (thangnc) [reply src/main/java/org/embulk/output/mailchimp/MailChimpClient.java]: @sakama noted and fixed

**Comment by thangnc** [comment]
@sakama PTAL

**Review by sakama — APPROVED** [review]
LGTM

**Pull request merged**